test/pytest/test_qkeras.py

```diff
@@ -229,6 +229,45 @@ def test_quantizer(randX_1000_1, quantizer, backend):
   # Goal is to get it passing with all equal
   np.testing.assert_array_equal(y_qkeras, y_hls4ml)
 
+def sigmoid(x):
+  return 1/(1 + np.exp(-x))
+
+@pytest.mark.parametrize('quantizer_pars', [(quantized_tanh(4), np.tanh, 0.07),
+                                       (quantized_tanh(8), np.tanh, 0.012),
+                                       (quantized_tanh(10), np.tanh, 0.009),
+                                       (quantized_sigmoid(3), sigmoid, 0.066),
+                                       (quantized_sigmoid(7), sigmoid, 0.009),
+                                       (quantized_sigmoid(9), sigmoid, 0.006)])
+@pytest.mark.parametrize('backend', ['Vivado', 'Quartus'])
+def test_quantizer_alt(randX_1000_1, quantizer_pars, backend):
+  '''
+  Test a single quantizer as an Activation function.
+  Checks the type inference through the conversion is correct without just
+  using the same logic.
+  '''
+  quantizer = quantizer_pars[0]
+  func = quantizer_pars[1]
+  tol = quantizer_pars[2]
+  X = randX_1000_1
+  X = np.round(X * 2**10) * 2**-10 # make it an exact ap_fixed<16,6>
+  model = Sequential()
+  model.add(QActivation(input_shape=(1,), activation=quantizer, name='quantizer'))
+  model.compile()
+
+  config = hls4ml.utils.config_from_keras_model(model, granularity='name')
+
+  output_dir = str(test_root_path / 'hls4mlprj_qkeras_quantizer_alt_{}_{}_{}'.format(quantizer.__class__.__name__,
+                                                            quantizer.bits, backend))
+  hls_model = hls4ml.converters.convert_from_keras_model(model,
+                                                       hls_config=config,
+                                                       output_dir=output_dir,
+                                                       backend=backend)
+  hls_model.compile()
+
+  y_calc = func(X)
```

Comment:
Why compare to a floating point version of the function on not the quantized one here?
i.e. you could do `y_calc = quantizer(X)`.
And then, on a related note, is there a problem just including these quantizers in `test_quantizer`?

Reply:
The reason for the different test is to do the different comparison. You have to do a try catch on the integer field, but in general, even if doing `use_qkeras_tanh=True` there were occasional differences that I could not solve by setting rounding and saturation modes. I can look more.

Reply:
I think it's quite likely that the `sigmoid` and `tanh` C++ needs a refresh to get bit-exactness wrt QKeras, they both have some hard coded values still (the `init_{sigmoid | tanh}_table` functions in particular)

Reply:
This has changed so I will resolve this conversion.